pyccl/pk2d.py

```diff
@@ -146,36 +152,37 @@ def __init__(self, *, a_arr=None, lk_arr=None, pk_arr=None,
 
     @classmethod
     def from_function(cls, pkfunc, *, is_logp=True,
-                      spline_params=spline_params,
+                      spline_par=None,
```

Comment:
Don't do that, it will be super confusing for people to refer to it with 2 different names.
What you need to do is remove `spline_params` from the imports at the top, and replace the lines in the function with:

```python
if spline_params is None:
    from . import spline_params
```
Much neater, and you get to keep the same name. We use `params` to refer to parameters everywhere in CCL. Let's be consistent.

Reply:
Yep, done.